### PR TITLE
Keyboard is fixed.

### PR DESCRIPTION
**Pull request by dmitry-zezik**

### CODE REVIEW

File: src/resources.qrc [hunk 1]
```diff
@@ -147,6 +147,7 @@
         <file>qml/components/TextScrollView.qml</file>
         <file>qml/components/AttachmentPicker.qml</file>
         <file>qml/components/LogControl.qml</file>
+	<file>qml/components/KeyboardHandler.qml</file>
```
Comment (SergeySeroshtan) [inline src/resources.qrc:150]: Fix indentation.

File: src/qml/components/KeyboardHandler.qml [hunk 2]
```diff
@@ -0,0 +1,107 @@
+import QtQuick 2.12
```
Comment (fpohtmeh-github) [inline src/qml/components/KeyboardHandler.qml:1]: as discussed, lets use 2.15 everywhere

File: src/qml/MainView.qml [hunk 3]
```diff
@@ -10,13 +10,12 @@ Control {
...
 
     property var attachmentPreview: undefined
-    property int keyboardHeight: 0
     readonly property var manager: app.stateManager
 
     RowLayout {
         anchors {
             fill: parent
-            bottomMargin: (logControl.visible ? logControl.height : 0) + keyboardHeight
+            bottomMargin: keyboardHandler.keyboardHeight
```
Comment (fpohtmeh-github) [inline src/qml/MainView.qml:18]: logControl logic is broken, restore deleted addition

File: src/qml/components/KeyboardHandler.qml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    function constKeyboardInit() {
+        let keyboard = Qt.inputMethod.keyboardRectangle
+        keyboardHandler.constKeyboardRectangle = Qt.rect(keyboard.x, keyboard.y, keyboard.width, keyboard.height)
+        keyboardHandler.constKeyboardYTranslate = keyboard.y - mainView.height
+    }
+
+    Connections {
+         target: visible ? Qt.inputMethod : null
```
Comment (fpohtmeh-github) [inline src/qml/components/KeyboardHandler.qml:83]: null is tricky, enabled = false is the correct way to disable connections

File: src/qml/components/KeyboardHandler.qml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+Item {
+    id: keyboardHandler
+
+    property real keyboardHeight: 0
+    property var constKeyboardRectangle
+    property real constKeyboardYTranslate: 0
+    property var currentKeyboardRectangle
+    property bool keyboardOpen: false
+    onKeyboardOpenChanged: {
+        if (!Platform.isDesktop) {
```
Comment (fpohtmeh-github) [inline src/qml/components/KeyboardHandler.qml:18]: I don't think this check is needed. Just disable Connections {} for Desktop platforrms

File: src/qml/MainView.qml (hunk 3 above)
Comment (dmitry-zezik) [reply src/qml/MainView.qml:18]: Ok

File: src/qml/components/KeyboardHandler.qml (hunk 2 above)
Comment (dmitry-zezik) [reply src/qml/components/KeyboardHandler.qml:1]: Ok

File: src/qml/components/KeyboardHandler.qml (hunk 4 above)
Comment (dmitry-zezik) [reply src/qml/components/KeyboardHandler.qml:83]: Ok

**Review by fpohtmeh-github — APPROVED** [review]

**Pull request merged**